### ISSUE TITLE
How to reload my trained model and retrain the model with new dataset?

### ISSUE THREAD

**Issue by easy1one** [googlecodelabs/tensorflow-for-poets-2#86]
Hi, I am new in the Machine Learning world! 

I have successfully created a model for Android. And then I want to retrain the model with new dataset (based on same classes; NOT transfer training). According to some internet resources, I probably need to use tf.Saver to save check points and meta model. For that, do I need to modify train.py script to keep "layer_weights = tf.Variable(initial_value, name='final_weights')" and "layer_biases = tf.Variable(tf.zeros([class_count]), name='final_biases')"  ?
If so, the problem is that how can I reload the model with saved .ckpt ad .meta model to retrain it? Can I reuse the retrain.py to retrain the model with saved variables ad model?

**Comment by easy1one** [comment]
UPADATE

I added global variables and tf.train.saver() in the main like following 

`
   # +++++++++++++++++++++++++++++++
   model_path = "./tmp/model.ckpt"
   save_path = "./tmp/model.ckpt"
   # +++++++++++++++++++++++++++++++

    ...

   def main(_):
   ....
   # Set up all our weights to their initial default values.
    init = tf.global_variables_initializer()
    sess.run(init)

    # +++++++++++++++++++++++++++++++++++++++++++++++++++
    # 'Saver' op to save and restore all the variables
    saver = tf.train.Saver()
    # # Restore model weights from previously saved model
    # saver.restore(sess, model_path)
    # print("Model restored from file: %s" % model_path)
    # +++++++++++++++++++++++++++++++++++++++++++++++++++

   ....

   # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    # Save model weights to disk
    save_path = saver.save(sess, model_path)
    # print("Model saved in file: %s" % save_path)
    # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
   `

Finally I could reload my previous model and keep training the model. 
For my case, I split the 100% dataset to 80% and 20%. Then creat the model with 80% dataset with original retrain.py and then reload and retrain the model with 20% dataset. Then I got the following results. 
# 1-1. model_0.75_192_80_v2
### tensorflow: Final test accuracy = 49.4% (N=1631)
### evaluate.py: Accuracy: 0.536481, Cross Entropy: 1.5794

# 1-2. model_0.75_192_100_v2_full
### tensorflow: Final test accuracy = 74.9% (N=447)
### evaluate.py: Accuracy: 0.727161, Cross Entropy: 0.920948


To test the model, I created the model with 100% dataset with original retrain.py then I got the following result.
# 1-3. model_0.75_192_100_v2_100
### tensorflow: Final test accuracy = 74.3% (N=2088)
### evaluate.py: Accuracy: 0.773145, Cross Entropy: 0.587493


First line (### tensorflow: Final test accuracy ) is from the last line after running retrain.py.
Second line (### evaluate.py: Accuracy: ... , Cross Entropy) is from the result after running evaluate.py with the models.

Another question is why the accuracy result is different between two?
And ThesorBoard has totally different results from these result.
For example, 
# 1-1. model_0.75_192_80_v2
### TensorflowBoard[last point]: (train)0.81/(validation)0.69

# 1-2. model_0.75_192_100_v2_full
### TensorflowBoard [last point]: (train)0.95/(validation)0.60

# 1-3. model_0.75_192_100_v2_100
### TensorflowBoard [last point]: (train)0.78/(validation)0.67

What is the correct accuracy ?


Thanks in advance! 👍
